packages/finance/src/liability/mod.rs

```diff
@@ -91,18 +91,27 @@ impl Liability {
         self.recalc_time
     }
 
-    pub fn init_borrow_amount<P>(&self, downpayment: P) -> P
+    pub fn init_borrow_amount<P>(&self, downpayment: P, max_ltv: Option<Percent>) -> P
     where
         P: Percentable,
     {
         use crate::fraction::Fraction;
+
+        debug_assert!(self.initial > Percent::ZERO);
         debug_assert!(self.initial < Percent::HUNDRED);
 
-        // borrow = init%.of(borrow + downpayment)
-        // (100% - init%).of(borrow) = init%.of(downpayment)
-        // borrow = init% / (100% - init%) * downpayment
-        let ratio = Rational::new(self.initial, Percent::HUNDRED - self.initial);
-        ratio.of(downpayment)
+        #[cfg(debug_assertions)]
+        if let Some(max_ltv) = max_ltv {
+            debug_assert!(max_ltv > Percent::ZERO);
+        }
```

Comment:
agreed not to add artificial limits, remove it and add a test with `max_ltv = Zero`